docs/adr/0016-publishing-times-and-change-history.md

```diff
@@ -0,0 +1,208 @@
+compatibility with Whitehall for migrating content due to it being a similar
+model.
+
+This negative is that it increases Content Publisher's responsibilities
+to storing time and change history. It also means that prior to publishing
+content Content Publisher would need to update timestamps in the Publishing API
+which adds an additional API call.
+
+#### 3. Adapt Publishing API to support an altering change history
+
+We considered modifying the Publishing API to allow the altering of change
+histories. This would involve change notes having a particular id value that
+could be used to identify them in requests.
+
+As a positive, this allows each change note to be treated as a separate concept
+to content and, compared to other approaches, allows a reduction in duplicate
+data shared between different editions of content.
+
+A negative of this is that it requires additional development and complexity
+for the Publishing API with a potential migration needed for existing content.
+It also presented a number of complications over whether change history would
+be part of a draft/live workflow, since content already has this concern.
+
+## Decision
+
+We decided to take the approach of storing times and change history in Content
+Publisher.
+
+We felt that this was the most pragmatic approach as it didn't require
+additional development to the Publishing API and was the most compatible with
+Whitehall's approach to change history. The increased control this offered
+was also consistent with a potential future feature of allowing change history
+to be modified via a user interface.
+
+We concluded that the recommended approach for change history in the Publishing
+API was ultimately flawed and should be reconsidered as a recommendation. Our
+concerns were:
+
+- all publishing applications, with the exception of Specialist Publisher,
+  sync their data to Publishing API with little regard to maintaining a
+  consistent history, which is a reversal from the
+  intention that Publishing API stores the history (an idea that was never
+  fully realised);
+- change history is the lone outlier as user visible data on GOV.UK that a
+  publishing application cannot fix in case of any issues;
+- the Publishing API ChangeNote model was designed to be append-only and
+  is therefore incompatible with the ability to change the time that content
+  is marked as first published (as the change history will disagree with this
+  value) or to resolve any mistakes.
+
+## Status
+
+Accepted
+
+## Consequences
+
+Content Publisher will store all the information needed to present
+`first_published_at`, `public_updated_at` and change history to the Publishing
+API. In doing so Content Publisher will be considered the canonical source
+of this data. This data will be reflected by non user editable data of
+`first_published_at` on a Document model and `published_at` on an Edition model,
+which reflects the times that Content Publisher first sent content to the
```

Comment:
Would this actually be "the times that Content Publisher published the content"?

Reply:
Thanks